core/src/main/java/org/apache/calcite/config/CalciteSystemProperty.java

```diff
@@ -124,12 +124,21 @@
   /**
    * Whether to enable top-down optimization.
    *
-   * <p>Note: Enabling top-down optimization will automatically disable
-   * the use of AbstractConverter and related rules.</p>
+   * <p>Note: Enabling top-down optimization will automatically enable
+   * top-down trait propagation.</p>
    */
   public static final CalciteSystemProperty<Boolean> TOPDOWN_OPT =
       booleanProperty("calcite.planner.topdown.opt", false);
 
+  /**
+   * Whether to enable top-down trait propagation.
+   *
+   * <p>Note: Enabling top-down optimization will automatically disable
+   * the use of AbstractConverter and related rules.</p>
+   */
+  public static final CalciteSystemProperty<Boolean> TOPDOWN_TRAITS =
+      booleanProperty("calcite.planner.topdown.traits", false);
```

Comment:
Why adding this new flag to control traits propagation?

Reply:
because we can enable trait propagation even though top down rule applying is disabled

Reply:
Discussed with Haisheng, I will remove this flag and combine all top-down optimization capability together.